evap/grades/templates/grades_semester_view.html

```diff
@@ -11,78 +11,79 @@
-                                                        <li><a href="{% url "grades:upload_grades" semester.id course.id %}?final=true">{% trans "Final grades" %}</a></li>
-                                                    </ul>
+                                            <a onclick="show_confirmNouploadModal({{ course.id }}, '{{ course.name|escapejs }}');" class="btn btn-sm btn-secondary {{ disable_if_archived }}" data-toggle="tooltip" data-placement="top" title="{% trans 'Confirm that final grades have been submitted but will not be uploaded.' %}"><span class="fa fa-times"></span></a>
+                                            {% if semester.is_archived %}
+                                                <a class="btn btn-sm btn-dark disabled"><span class="fa fa-upload"></span></a>
+                                            {% else %}
+                                                <div class="btn-group" role="group" data-toggle="tooltip" data-placement="top" title="{% trans 'Upload grade document' %}">
+                                                    <a id="btnUpload{{ course.id }}" role="button" class="btn btn-sm btn-dark dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span class="fa fa-upload"></span></a>
+                                                    <div class="dropdown-menu" aria-labelledby="btnUpload{{ course.id }}">
+                                                        <a class="dropdown-item" href="{% url 'grades:upload_grades' semester.id course.id %}">{% trans "Midterm grades" %}</a>
+                                                        <a class="dropdown-item" href="{% url 'grades:upload_grades' semester.id course.id %}?final=true">{% trans "Final grades" %}</a>
+                                                    </div>
                                                 </div>
-                                            </div>
+                                            {% endif %}
                                         {% else %}
-                                            <span data-toggle="tooltip" data-placement="top" title="{% trans "Change grade document" %}">
-                                                <a class="btn btn-sm btn-default" href="{% url "grades:course_view" semester.id course.id %}" {{ disable_if_archived }}>
-                                                    <span class="glyphicon glyphicon-pencil"></span>
-                                                </a>
-                                            </span>
+                                            <a href="{% url 'grades:course_view' semester.id course.id %}" class="btn btn-sm btn-light {{ disable_if_archived }}" data-toggle="tooltip" data-placement="top" title="{% trans 'Change grade document' %}"><span class="fa fa-pencil"></span></a>
```

Comment:
you left disable_if_archived on a `a href` here but not above

Reply:
see above